### DISCUSSION TITLE
skynet的actor问题

### DISCUSSION THREAD

**Discussion by cppbeliever** [cloudwu/skynet#1859] [General]
都说actor之间不共享内存，那么，请问，actor的mailbox是不是共享内存，是不是在多线程环境下运行时需要加锁？  

比如：actor_B 和 actor_C 这两个线程同时向actor_A的mailbox发送异步消息，并且actor_A线程从mailbox取出消息进行处理。三个线程同时运行，请问，actor_A的mailbox是不是加锁？   

我个人认为，肯定要加锁，只是对mailbox的操作粒度很小，加的是自选锁而已。

我对actor的理解是小白，求解惑。

**Comment by sanergogogo** [comment]
每个actor就相当于一个单进程单线程的环境。
所以你觉得需要加锁吗？

**Reply by terry8210** [reply]
底层处理消息的同步和锁的问题，直接send 和 call就行了

**Reply by cppbeliever** [reply]
所以，还是有锁，只不过在上层的业务层面没有锁，在底层的消息分发机制中还是有锁的。
